Dockerfile.base

```diff
@@ -1,64 +1,38 @@
 # syntax=docker/dockerfile:1
 
 # BASE docker image for music assistant container
-# Based on Debian Trixie (testing) because we need a newer version of ffmpeg (and snapcast)
-# TODO: Switch back to regular python stable debian image + manually build ffmpeg and snapcast ?
-FROM debian:trixie-slim
-
-ARG TARGETPLATFORM
+#
+# It should contain ONLY packages we want to deploy in the final image, not
+# packages needed to build. Build dependencies should be installed in the
+# "builder" image in the main Dockerfile.
 
+FROM python:3.12-alpine3.21
 
 RUN set -x \
-    && apt-get update \
-    && apt-get install -y --no-install-recommends \
+    && apk add --no-cache \
         ca-certificates \
-        curl \
-        git \
-        wget \
+        jemalloc \
         tzdata \
-        python3 \
-        python3-venv \
-        python3-pip \
-        libsox-fmt-all \
-        libsox3 \
-        ffmpeg \
         sox \
-        openssl \
         cifs-utils \
-        libnfs-utils \
-        libjemalloc2 \
-        snapserver \
-    # cleanup
-    && rm -rf /tmp/* \
-    && rm -rf /var/lib/apt/lists/*
+        snapcast \
+        libnfs \
+        openssl-dev  # needed for airplay
 
+# Get static ffmpeg builds from https://hub.docker.com/r/mwader/static-ffmpeg/
```

Comment:
How good are these builds ? I mean, the core alpine team struggles with getting ffmpeg 7.1 to cooperate with musl so what makes this work magically ?

Reply:
I had exactly the same question, I think the answer (judging from their [Dockerfile](https://github.com/wader/static-ffmpeg/blob/master/Dockerfile)) is that they actually use glibc (or at least partially) instead of musl, but I am not sure.

Now the quality of the builds is hard to assess, but the image is reasonably popular (500K pulls) which is a good sign. I'd say better than building ourselves in any case.

I guess what's important is to test whether the issues from music-assistant/support/issues/3257 are still present with these builds or not.

Reply:
Hey lurking around a bit 👋 the builds only uses musl (maybe you confused glib and glibc?). The build do some patching (lib build fixes that i usually also upstream) and tweaking (static PIE, stack size etc) to work better with musl and behave more similar to glibc. I've use a similar build to this at work where is has decoded a lot of random media from the internet for years and have had very few problems.

If you want an even smaller ELF-binary i would suggest building ffmpeg yourself with less dependencies and features enabled.